### ISSUE TITLE
Support for Export Layers plugin

### ISSUE THREAD

**Issue by candideu** [glimpse-editor/Glimpse#460]
Would it be possible to support the Export Layers plugin, which is a plug-in that exports layers as separate images? : https://khalim19.github.io/gimp-plugin-export-layers/

**Comment by TrechNex** [comment]
We do not include Python 2 runtime components with Windows, so will be unable to do it for that operating system. There should be no problems supporting it on Linux though.

Initially we can document the steps needed to install it in our wiki (as we do for other plugins), but could look into pre-bundling in future versions if there's enough people who want us to do it. 👍

**Comment by candideu** [comment]
> 
> 
> We do not include Python 2 runtime components with our Windows port, so will be unable to do it for that operating system. There should be no problems supporting it on Linux though.
> 
> Initially we can document the steps needed to install it in our wiki (as we do for other plugins), but could look into pre-bundling in future versions if there's enough people who want us to do it. 👍

I'm running on Windows 10. Is there any reason why the Python 2 components aren't included?

I would personally love to have this plugin or any other that would allow users to export each layer separately as a PNG or JPEG.

**Comment by TrechNex** [comment]
@candideu there's a few 😅 

a) There's a known bug that prevents us from building the main binary for Windows with Python support using MSYS2 (our build system of choice). #178 
b) Python 2 is now "End of Life", so it will not be receiving any further security updates from the Python development team. That is not a problem on Linux because we can securely sandbox the application, but that is not something we can do on Windows currently (Although we are looking into it here: #322)
c) The application loads a lot faster on Windows without Python filters and plugins, which means users are less likely to try and launch Glimpse Image Editor multiple times before the splash screen appears
d) The feedback we had from most Windows users is they don't actually notice the things that are missing unless they're specifically pointed out, which raises questions about how important to the user experience they were to begin with

The GNU Image Manipulation Program developers have introduced Python 3 support on their master branch, so if that is ever backported in a future 2.10.x release, then we would eventually inherit it (assuming MSYS2 is also updated) in a future Glimpse Image Editor release. We could then fork plugins like these and "upgrade" them from Python 2 to Python 3, as there is no backwards-compatibility by default.

Our understanding though is that the GNU Image Manipulation Program developers do not plan to backport Python 3 support to 2.10.x at the present time, and instead plan to wait for their future 3.0.x release which does not have a fixed release date. That impacts us as a downstream fork, and it also impacts Linux users running recent distributions that have deprecated Python 2 support.

**Comment by ryliejamesthomas** [comment]
(just chippig-in to say I have found this plugin very useful in the past for game-making related stuff)

**Comment by candideu** [comment]
> 
> 
> @candideu there's a few 😅
> 
> a) There's a known bug that prevents us from building the main binary for Windows with Python support using MSYS2 (our build system of choice). #178
> b) Python 2 is now "End of Life", so it will not be receiving any further security updates from the Python development team. That is not a problem on Linux because we can securely sandbox the application, but that is not something we can do on Windows currently (Although we are looking into it here: #322)
> c) The application loads a lot faster on Windows without Python filters and plugins, which means users are less likely to try and launch Glimpse Image Editor multiple times before the splash screen appears
> d) The feedback we had from most Windows users is they don't actually notice the things that are missing unless they're specifically pointed out, which raises questions about how important to the user experience they were to begin with
> 
> The GNU Image Manipulation Program developers have introduced Python 3 plugin support on their master branch, so if that is ever backported in a future 2.10.x release, then we would eventually inherit it (assuming MSYS2 is also updated) in a future Glimpse Image Editor release. We could then fork plugins like these and "upgrade" them from Python 2 to Python 3, as there is no backwards-compatibility by default.
> 
> Our understanding though is that the GNU Image Manipulation Program developers do not plan to backport Python 3 plugin support to 2.10.x at the present time, and instead plan to wait for their future 3.0.x release which does not have a fixed release date. That impacts us as a downstream fork, and it also impacts Linux users running recent distributions that have deprecated Python 2 support.
> 
> **tl;Dr:** Python support is a bit of a mess right now because of Python 2 being deprecated across different software/operating systems and upstream delaying the introduction of Python 3 support for various reasons. We're doing the best we can under the circumstances. 🙂

Thank you so much for taking the time to explain it! 🙂
